src/audio/module_adapter/module/generic.c

```diff
@@ -318,7 +329,18 @@ static int free_contents(struct processing_module *mod, struct module_resource *
 
 	switch (container->type) {
 	case MOD_RES_HEAP:
+#if CONFIG_SOF_PACOVR
+	/* do we need to use the scratch heap or the batch heap? */
```

Comment:
I do not think this would work. Its not guaranteed that memory allocated in initialization phase is freed in also in initialization phase. For the moment there are modules freeing memory at initialized phase, that was allocated during init pahse. 

It should not be too hard to write a single pacovr_free() that knows from the pointer's memory range if its allocated scratch or batch (=> dynamic or static) heap.